src/metabase/lib/join.cljc

```diff
@@ -287,11 +289,11 @@
   "Create an MBQL join map from something that can conceptually be joined against. A `Table`? An MBQL or native query? A
   Saved Question? You should be able to join anything, and this should return a sensible MBQL join map."
   ([joinable]
-   (join-clause-method joinable))
+   (-> (join-clause-method joinable)
+       (u/assoc-default :fields :all)))
```

Comment:
Unrelated fix, `:fields :all` wasn't being added to join clauses by default if you called the 1 arity

Reply:
It looks now like it might not be called for with the 2-arity?

Reply:
The two-arity calls the one-arity